plugins/otter-pro/inc/plugins/form/class-form-submissions.php

```diff
@@ -0,0 +1,983 @@
+<?php
+/**
+ * Form Block Responses Storing.
+ *
+ * @package ThemeIsle\OtterPro\Plugins\Form
+ */
+
+namespace ThemeIsle\OtterPro\Plugins\Form;
+
+use ThemeIsle\GutenbergBlocks\Integration\Form_Data_Request;
+use ThemeIsle\GutenbergBlocks\Server\Form_Server;
+use ThemeIsle\OtterPro\Plugins\License;
+use WP_Post;
+use WP_Query;
+
+/**
+ * Class Form_Block
+ */
+class Form_Block_Emails_Storing {
+	/**
+	 * Form record post type.
+	 */
+	const FORM_RECORD_TYPE = 'otter_form_record';
+
+	/**
+	 * Form record meta key.
+	 */
+	const FORM_RECORD_META_KEY = 'otter_form_record_meta';
+
+	/**
+	 * The main instance var.
+	 *
+	 * @var Form_Block
+	 */
+	public static $instance = null;
+
+	/**
+	 * Initialize the class
+	 */
+	public function init() {
+		if ( ! License::has_active_license() ) {
+			return;
+		}
+
+		add_action( 'init', array( $this, 'create_form_records_type' ) );
+		add_action( 'admin_init', array( $this, 'set_form_records_cap' ), 10, 0 );
+		add_action( 'otter_form_after_submit', array( $this, 'store_form_record' ) );
+		add_action( 'admin_head', array( $this, 'add_style' ) );
+
+		// Customize the wp_list_table.
+		add_filter( 'manage_' . self::FORM_RECORD_TYPE . '_posts_columns', array( $this, 'form_record_columns' ) );
+		add_filter( 'manage_edit-' . self::FORM_RECORD_TYPE . '_sortable_columns', array( $this, 'form_record_sortable_columns' ) );
+		add_filter( 'manage_' . self::FORM_RECORD_TYPE . '_posts_custom_column', array( $this, 'form_record_column_values' ), 10, 2 );
+		add_filter( 'bulk_actions-edit-' . self::FORM_RECORD_TYPE, array( $this, 'form_record_bulk_actions' ) );
+		add_filter( 'handle_bulk_actions-edit-' . self::FORM_RECORD_TYPE, array( $this, 'handle_form_record_bulk_actions' ), 0, 3 );
+
+		add_filter( 'post_row_actions', array( $this, 'form_record_row_actions' ), 10, 2 );
+		add_action( 'restrict_manage_posts', array( $this, 'form_record_add_filters' ) );
+		add_filter( 'parse_query', array( $this, 'form_record_filter_query' ) );
+		add_action( 'transition_post_status', array( $this, 'transition_draft_to_read' ), 10, 3 );
+
+		// Implement row actions behaviour.
+		add_action( 'admin_action_row-read', array( $this, 'read_otter_form_record' ) );
+		add_action( 'admin_action_row-unread', array( $this, 'unread_otter_form_record' ) );
+		add_action( 'admin_action_edit', array( $this, 'mark_read_on_edit' ) );
+
+		// Manage meta boxes.
+		add_action( 'add_meta_boxes', array( $this, 'add_form_record_meta_box' ) );
+		add_action( 'admin_menu', array( $this, 'handle_admin_menu' ) );
+		add_action( 'save_post', array( $this, 'form_record_save_meta_box' ), 10, 2 );
+	}
+
+	/**
+	 * Create custom post type for form records.
+	 *
+	 * @return void
+	 */
+	public function create_form_records_type() {
+		register_post_type(
```

Comment:
One major issue right now that I see is that any user even with the role of Contributor can access this Dashboard. We should limit it to only people with admin capabilities.

Reply:
If you're using a role lower than Admin, you can visit the Dashboard by accessing the URl directly.